hypertrace-trace-enricher/hypertrace-trace-enricher-api/src/main/java/org/hypertrace/traceenricher/trace/util/GraphBuilderUtil.java

```diff
@@ -0,0 +1,40 @@
+package org.hypertrace.traceenricher.trace.util;
+
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GraphBuilderUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(GraphBuilderUtil.class);
+
+  /**
+   * optimistic method of comparing two trace for considering rebuilding of entire graph structure.
+   */
+  static boolean isSameStructuredTrace(StructuredTrace cachedTrace, StructuredTrace trace) {
```

Comment:
the method name could be misleading, we are trying to check if the components in the Trace are same as previous or not. Trace might have changed due to event enrichment or trace enrichment.